dask/dataframe/tests/test_shuffle.py

```diff
@@ -1007,3 +1008,30 @@ def test_set_index_overlap():
     a = a.set_index("key", sorted=True)
     b = a.repartition(divisions=a.divisions)
     assert_eq(a, b)
+
+
+def test_shuffle_hlg_layer():
+    # This test checks that the `ShuffleLayer` HLG Layer
+    # is used (as expected) for a multi-stage shuffle.
+    ddf = dd.from_pandas(
+        pd.DataFrame({"a": np.random.randint(0, 10, 100)}), npartitions=10
+    )
+    ddf_shuffled = ddf.shuffle("a", max_branch=3, shuffle="tasks")
+    keys = [(ddf_shuffled._name, i) for i in range(ddf_shuffled.npartitions)]
+
+    # Make sure HLG culling reduces the graph size
+    dsk = ddf_shuffled.__dask_graph__()
+    dsk_culled = dsk.cull(set(keys))
+    assert len(dsk_culled) < len(dsk)
+    assert isinstance(dsk_culled, dask.highlevelgraph.HighLevelGraph)
+
+    # Check HLG layers beginning with "shuffle-"
+    for name, layer in dsk.layers.items():
+        if name.startswith("shuffle-"):
+            assert isinstance(layer, dd.shuffle.ShuffleLayer)
+
+    # Since we already culled the HLG,
+    # culling the dictionary should not change the graph
+    dsk_dict = dict(dsk)
+    dsk_dict_culled, _ = cull(dsk_dict, keys)
+    assert len(dsk_dict_culled) < len(dsk_dict)
```

Comment:
My understanding about what we're trying to test here could be wrong, but based on the comment above I would have expected us to cull the already culled high-level graph, `dsk_culled`, and then assert that the the graph is the same before and after culling (since it has already been culled earlier):

```suggestion
    dsk_dict = dict(dsk_culled)
    dsk_dict_culled, _ = cull(dsk_dict, keys)
    assert dsk_dict_culled == dsk_dict
```

Reply:
good catch!  My mistake.